frontend/app/src/utils/message-handling.ts

```diff
@@ -73,3 +76,15 @@ export const handlePremiumStatusUpdate = (
 
   return null;
 };
+
+export const handleLoginStatus = (
+  message: WebsocketMessage<SocketMessageType>
+) => {
+  const store = useSessionAuthStore();
+  const { logged } = storeToRefs(store);
+  const { updateLoginStatus } = store;
+  const data = LoginStatusData.parse(convertKeys(message.data, true, false));
+  if (!get(logged)) {
+    updateLoginStatus(data);
+  }
+};
```

Comment:
I think it would maybe make sense to put this functionality inside the store instead